src/Filter/LimitFilter.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Filter;
+
+use ApiPlatform\Doctrine\Orm\Filter\AbstractFilter;
+use ApiPlatform\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Metadata\Operation;
+use Doctrine\ORM\QueryBuilder;
+use Symfony\Component\PropertyInfo\Type;
+
+/**
+ * Filter to limit the number of results in the result set.
+ */
+final class LimitFilter extends AbstractFilter
+{
+    /**
+     * {@inheritDoc}
+     */
+    protected function filterProperty(string $property, $value, QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, Operation $operation = null, array $context = []): void
+    {
+        $queryBuilder->setMaxResults($value);
```

Comment:
Consider having the application define an absolute max so that a client cannot do `?max=100000`. Could just be
```
$value = self::MAX_ALLOWED_RESULTS < $value ? self::MAX_ALLOWED_RESULTS : $value;
```